token-client-core/src/main/java/no/nav/security/token/support/client/core/ClientProperties.java

```diff
@@ -0,0 +1,25 @@
+package no.nav.security.token.support.client.core;
+
+import lombok.Data;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.util.List;
+
+@Data
+public class ClientProperties {
+    @NotNull
+    private URI resourceUrl;
+    @NotNull
+    private URI tokenEndpointUrl;
+    @NotEmpty
+    private String clientId;
+    @NotEmpty
+    private String clientSecret;
+    private String clientAuthMethod = "client_secret_basic";
```

Comment:
Burde denne eventuelt også vært en enum for å vise klarere hva som støttes?   Hvorfor er det av og til basic og av og til post, forresten? Forskjellig implementasjon hos id-providerne?   Det er denne som også da vil være "jwt" hvis man bruker jwt/cert ja?

Reply:
Kunne kanskje vært en enum ja gjorde det enkelt nå, men er fort gjort å smelle inn en enum. Det er idproviderne som egentlig styrer hva man skal bruke gjennom `token_endpoint_auth_methods_supported` i metadataen sin. for jwt cert skal den vel være `private_key_jwt`. Har ikke lagt inn støtte ennå, men har kode liggende, tenkte å lage det som en egen PR.

Reply:
Ok. Jepp. Fort gjort å fikse etterhvert som du sier. Blir jo "breaking changes" da, om enn ikke veldig kompliserte :)